<launch>
  <arg name="ro_path" default="$(find reachable_occupancy)"/>
  <arg name="mvmt_csv" default="$(arg ro_path)/src/scripts/tests/simple_skeleton_mvmt.csv"/>
  <arg name="arm_csv" default="$(arg ro_path)/src/scripts/tests/arms.csv"/>
  <arg name="pos_walking" default="false"/>
  <arg name="autostop" default="false"/>
  <arg name="frame_stop" default="253"/>
  <arg name="only_skel" default="false"/>
  <arg name="frame_id" default="map"/>
  <arg name="def_delta_t" default="0.02"/>
  <arg name="use_arm" default="false"/>
  <arg name="draw_human" default="false"/>
  <arg name="only_points" default="false"/>
  <arg name="sensor_noVelocity" default="false"/>
  <arg name="visualization_accel" default="false"/>
  <arg name="visualization_vel" default="false"/>
  <arg name="visualization_pos" default="false"/>
  <arg name="visualization_all_art" default="false"/>
  <arg name="visualization_ped" default="false"/>
  <arg name="pedestrian_mode" default="false"/>
  <arg name="articulated_mode" default="false"/>
  <arg name="steps" default="4"/>
  <arg name="ped_vel_alpha" default="0.6"/>
  <arg name="ped_accel_alpha" default="0.6"/>
  <arg name="art_pos_alpha" default="0.6"/>
  <arg name="art_vel_alpha" default="0.6"/>
  <arg name="art_accel_alpha" default="0.6"/>

  <rosparam file = "$(arg ro_path)/src/config/reachable_occupancy.yaml" />
  <param name="ro_path" type="str" value="$(arg ro_path)"/>
  <group if="$(arg use_arm)">
    <param name="csv_path" type="str" value="$(arg arm_csv)"/>
  </group>
  <group unless="$(arg use_arm)">
    <param name="csv_path" type="str" value="$(arg mvmt_csv)"/>
  </group>
  <param name="draw_human" type="bool" value="$(arg draw_human)"/>
  <param name="autostop" type="bool" value="$(arg autostop)"/>
  <param name="pos_walking" type="bool" value="$(arg pos_walking)"/>
  <param name="frame_stop" type="int" value="$(arg frame_stop)"/>
  <param name="only_skel" type="bool" value="$(arg only_skel)"/>
  <param name="frame_id" type="str" value="$(arg frame_id)"/>
  <param name="def_delta_t" type="double" value="$(arg def_delta_t)"/>
  <param name="only_points" type="bool" value="$(arg only_points)"/>
  <param name="sensor_noVelocity" type="bool" value="$(arg sensor_noVelocity)"/>
  <group if="$(arg visualization_all_art)">
    <param name="visualization_accel" type="bool" value="true"/>
    <param name="visualization_vel" type="bool" value="true"/>
    <param name="visualization_pos" type="bool" value="true"/>
  </group>
  <group unless="$(arg visualization_all_art)">
    <param name="visualization_accel" type="bool" value="$(arg visualization_accel)"/>
    <param name="visualization_vel" type="bool" value="$(arg visualization_vel)"/>
    <param name="visualization_pos" type="bool" value="$(arg visualization_pos)"/>
  </group>
  <param name="visualization_ped" type="bool" value="$(arg visualization_ped)"/>
  <param name="pedestrian_mode" type="bool" value="$(arg pedestrian_mode)"/>
  <param name="articulated_mode" type="bool" value="$(arg articulated_mode)"/>
  <param name="steps" type="int" value="$(arg steps)"/>
  <param name="ped_vel_alpha" type="double" value="$(arg ped_vel_alpha)"/>
  <param name="ped_accel_alpha" type="double" value="$(arg ped_accel_alpha)"/>
  <param name="art_pos_alpha" type="double" value="$(arg art_pos_alpha)"/>
  <param name="art_vel_alpha" type="double" value="$(arg art_vel_alpha)"/>
  <param name="art_accel_alpha" type="double" value="$(arg art_accel_alpha)"/>


  <node name="rviz" pkg="rviz" type="rviz" args="-d $(arg ro_path)/rviz/ro_config_no-angle.rviz"/>
  <node pkg="reachable_occupancy" type="reachable_ocpc" name="occupancy_calculator" required="true" output="screen"/>
  <node pkg="reachable_occupancy" type="skeleton_publisher.py" name="demo_point_publisher" output="screen"/>
</launch>
